integrated_channels/integrated_channel/exporters/course_metadata.py

```diff
@@ -0,0 +1,104 @@
+    """
+
+    AVAILABILITY_CURRENT = 'Current'
+    AVAILABILITY_UPCOMING = 'Upcoming'
+    AVAILABILITY_ARCHIVED = 'Archived'
+
+    def __init__(self, user, enterprise_configuration):
+        """
+        Save the appropriate details for use elsewhere in the object.
+        """
+        super(CourseExporter, self).__init__(user, enterprise_configuration)
+        self.removed_courses_resolved = False
+        self.courses = []
+        client = EnterpriseApiClient(self.user)
+        enterprise_course_runs = client.get_enterprise_course_runs(self.enterprise_customer)
+        LOGGER.info('Retrieving course list for enterprise %s', self.enterprise_customer.name)
+        for course_run in enterprise_course_runs.values():
+            self.add_course_run(course_run)
+
+    def export(self):
+        """
+        Export the set of courses generically by encoding them in JSON format.
+        """
+        yield json.dumps(self.courses, sort_keys=True).encode('utf-8')
+
+    @property
+    def data(self):
+        """
+        Return blanket data by default.
+
+        If a course exporter requires data transformation, it should override this property.
+        """
+        return {}
+
+    def add_course_run(self, course_run):
+        """
+        Transform the details of a course run, and save it to our courses list.
+        """
+        transformed = self.transform(course_run)
+        LOGGER.info(
+            'Sending course with plugin configuration %s: %s',
+            self.enterprise_configuration,
+            json.dumps(transformed, indent=4),
+        )
+        self.courses.append(transformed)
+
+    def transform(self, course_run):
+        """
+        Parse the provided course into the format natively supported by the provider.
+        """
+        LOGGER.info('Processing course with ID %s', course_run['key'])
+        LOGGER.debug('Parsing course for %s: %s', self.enterprise_customer, json.dumps(course_run, indent=4))
+
+        # Add the enterprise customer to the course run details so it can be used in the data transform
+        course_run['enterprise_customer'] = self.enterprise_customer
+        transformed_data = {}
+        for key, transform in self.data.items():
+            transformed_data[key] = transform(course_run) if transform is not None else course_run.get(key)
+        return transformed_data
+
+    def resolve_removed_courses(self, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Default way to resolve the removal of courses.
+
+        This is implemented as such to allow certain integrated channels that do not really utilize
+        this functionality to just return the entire set of ready-to-transmit courses for auditing
+        purposes.
+
+        Note that these integrated channels are presumed to be handling updating their course
+        availability on upstream catalogs through some other technique, i.e. DELETE requests
```

Comment:
Degreed in this case (yeah, that also means we need to implement DELETE requests for Degreed if we plan on explicitly marking courses in upstream catalogs as inactive (same goes for removing learner completions); I think this would essentially remove them from the catalog entirely, which I'm not sure about doing. cc @douglashall) If we do need to remove them, that should be pretty quick to add to what we already have.

I've reworded things to make this clearer.

Reply:
> I've reworded things slightly to remove any confusion, but would like not to mention Degreed directly of course, to keep the concept generic.

Thanks! Yeah, I didn't mean to suggest that this docstring should mention Degreed/SAPSF; just wanted to confirm if my understanding of its contents was correct :)